src/m365/spo/commands/web/web-roleassignment-add.ts

```diff
@@ -115,15 +132,29 @@ class SpoWebRoleAssignmentAddCommand extends SpoCommand {
 
       if (args.options.upn) {
         args.options.principalId = await this.getUserPrincipalId(args.options);
-        await this.addRoleAssignment(logger, args.options);
       }
       else if (args.options.groupName) {
         args.options.principalId = await this.getGroupPrincipalId(args.options);
-        await this.addRoleAssignment(logger, args.options);
       }
-      else {
-        await this.addRoleAssignment(logger, args.options);
+      else if (args.options.entraGroupId || args.options.entraGroupName) {
+        if (this.verbose) {
+          await logger.logToStderr('Retrieving group information...');
+        }
+
+        let group: Group;
+        if (args.options.entraGroupId) {
+          group = await entraGroup.getGroupById(args.options.entraGroupId);
+        }
+        else {
+          group = await entraGroup.getGroupByDisplayName(args.options.entraGroupName!);
+        }
```

Comment:
similar like in the listItem PR we may just simplify this and remove the type. The type is returned by the method so it is applied allready no need to mention it I guess.

```suggestion
        const group = args.options.entraGroupId ? await entraGroup.getGroupById(args.options.entraGroupId) : await entraGroup.getGroupByDisplayName(args.options.entraGroupName!);
```